src/com/maddyhome/idea/vim/group/MotionGroup.java

```diff
@@ -118,257 +118,58 @@ public void turnOff() {
-      ExEntryPanel.getInstance().deactivate(false);
-    }
-
-    ExOutputModel.getInstance(editor).clear();
-
-    if (CommandState.getInstance(editor).getMode() == CommandState.Mode.VISUAL) {
-      CommandState.getInstance(editor).popState();
-    }
-
-    int start = editor.getSelectionModel().getSelectionStart();
-    int end = editor.getSelectionModel().getSelectionEnd();
-    if (start == end) return;
-
-    if (mode == CommandState.SubMode.VISUAL_LINE) {
-      end--;
-      endOff--;
-    }
-
-    if (end == startOff || end == endOff) {
-      int t = start;
-      start = end;
-      end = t;
-
-      if (mode == CommandState.SubMode.VISUAL_CHARACTER) {
-        start--;
-      }
-    }
-
-    MotionGroup.moveCaret(editor, editor.getCaretModel().getPrimaryCaret(), start);
-    toggleVisual(editor, 1, 0, mode);
-    MotionGroup.moveCaret(editor, editor.getCaretModel().getPrimaryCaret(), end);
-    KeyHandler.getInstance().reset(editor);
-  }
-
-  @NotNull
-  public TextRange getWordRange(@NotNull Editor editor, @NotNull Caret caret, int count, boolean isOuter,
-                                boolean isBig) {
-    int dir = 1;
-    boolean selection = false;
-    if (CommandState.getInstance(editor).getMode() == CommandState.Mode.VISUAL) {
-      if (CaretData.getVisualEnd(caret) < CaretData.getVisualStart(caret)) {
-        dir = -1;
-      }
-      if (CaretData.getVisualStart(caret) != CaretData.getVisualEnd(caret)) {
-        selection = true;
-      }
-    }
-
-    return SearchHelper.findWordUnderCursor(editor, caret, count, dir, isOuter, isBig, selection);
-  }
-
-  @Nullable
-  public TextRange getBlockQuoteRange(@NotNull Editor editor, @NotNull Caret caret, char quote, boolean isOuter) {
-    return SearchHelper.findBlockQuoteInLineRange(editor, caret, quote, isOuter);
-  }
-
-  @Nullable
-  public TextRange getBlockRange(@NotNull Editor editor, @NotNull Caret caret, int count, boolean isOuter, char type) {
-    return SearchHelper.findBlockRange(editor, caret, type, count, isOuter);
-  }
-
-  @Nullable
-  public TextRange getBlockTagRange(@NotNull Editor editor, @NotNull Caret caret, int count, boolean isOuter) {
-    return SearchHelper.findBlockTagRange(editor, caret, count, isOuter);
-  }
-
-  @NotNull
-  public TextRange getSentenceRange(@NotNull Editor editor, @NotNull Caret caret, int count, boolean isOuter) {
-    return SearchHelper.findSentenceRange(editor, caret, count, isOuter);
-  }
-
-  @Nullable
-  public TextRange getParagraphRange(@NotNull Editor editor, @NotNull Caret caret, int count, boolean isOuter) {
-    return SearchHelper.findParagraphRange(editor, caret, count, isOuter);
-  }
-
   /**
-   * This helper method calculates the complete range a motion will move over taking into account whether
-   * the motion is FLAG_MOT_LINEWISE or FLAG_MOT_CHARACTERWISE (FLAG_MOT_INCLUSIVE or FLAG_MOT_EXCLUSIVE).
+   * This helper method calculates the complete range a visual will move over taking into account whether
```

Comment:
Here and everywhere else in this file I believe "motion" has been changed to "visual" accidentally. Could you please revert it?

Reply:
I've found the bad commit and reverted changes according to them